tidy3d/web/api/autograd/autograd.py

```diff
@@ -529,26 +530,32 @@ def postprocess_fwd(
 
 def upload_sim_fields_keys(sim_fields_keys: list[tuple], task_id: str, verbose: bool = False):
     """Function to grab the VJP result for the simulation fields from the adjoint task ID."""
-    data_file = tempfile.NamedTemporaryFile(suffix=".hdf5")
-    data_file.close()
-    TracerKeys(keys=sim_fields_keys).to_file(data_file.name)
-    upload_file(
-        task_id,
-        data_file.name,
-        SIM_FIELDS_KEYS_FILE,
-        verbose=verbose,
-    )
+    handle, fname = tempfile.mkstemp(suffix=".hdf5")
+    os.close(handle)
+    try:
+        TracerKeys(keys=sim_fields_keys).to_file(fname)
+        upload_file(
+            task_id,
+            fname,
+            SIM_FIELDS_KEYS_FILE,
+            verbose=verbose,
+        )
+    finally:
+        os.unlink(fname)
```

Comment:
Looks good but can you explain why we couldn't just delete `data_file.name` with `os.unlink` or `shutil.rmtree` or something like that without changing the rest of the code?

Reply:
yeah that's fair, I only did this to be consistent to the way we're handling temp files in base.py, but I can change it

Reply:
No, that's fine, that's a good enough reason, I was just wondering.

But while you're at this could you actually browse through all the places we use `tempfile`? I think there may be others to fix too, like in the old adjoint plugin.

Reply:
ok yeah sounds good, will do

Reply:
yup same problem in adjoint plugin, everywhere else looks fine